src/uint/modular/constant_mod/const_pow.rs

```diff
@@ -33,6 +38,40 @@ impl<MOD: ResidueParams<LIMBS>, const LIMBS: usize> Residue<MOD, LIMBS> {
             phantom: core::marker::PhantomData,
         }
     }
+
+    /// Raises each `base` to its `exponent` power,
+    #[cfg(feature = "alloc")]
+    pub fn multi_exponentiate(
+        bases_and_exponents: Vec<(Self, Uint<LIMBS>)>,
```

Comment:
Isn't this a similar problem to your other PRs, where this code could potentially be heapless-friendly if you had a const generic parameter to size the temporary storage, and it might be nice to have both const generic and alloc-based APIs, so possibly a trait impl'd for multiple input types (where I think a slice would suffice here, since the types involved are all `Copy` anyway)

Reply:
hmm I kinda did that internally, but didn't expose. I will

Reply:
One problem that arises is the fact `pow_montgomery_form()` is `const`. This does not allow me to keep the allocation there and move the initialization to a `_generic()` helper function, since: ` mutable references are not allowed in constant functions`.

Reply:
Okay, seems fine for now then

Reply:
No please wait I'm close to have something

Reply:
@tarcieri Ok I'm done. The mutable reference forced me to have some shared functionality DRYed-out into a different function, but I'm overall happy with the result, and generally with this framework we came up with of doing a generic trait implemented both for `alloc` of dynamically sized slices and `no-alloc` for const-generic arrays.